libsrc/osic1p/cputc.s

```diff
@@ -36,10 +41,13 @@ L1:     cmp     #$0D            ; LF?
 cputdirect:
```

Comment:
I hadn't though about using macros to accomplish the same goal, and that's certainly an alternative that will result in slightly smaller code. I will try to change the implementation to macros.

Regarding the scrolling in conio I would prefer to leave that in place. This was actually the first contribution that I received from another user who needed it.

At the end of line the conio functions are supposed to wrap around to the next line, aren't they? So a comparison for end-of-line is necessary in any case. Scrolling at the end of the screen makes that behavior consistent. If the programmer does not rely on scrolling and takes care himself not to run over the end of the screen, the only speed impact will be one comparison for end-of-screen that never evaluates to true. I hope that this minimal speed impact is acceptable.

POSIX I/O is on the radar, and in my plan it will be based on the conio implementation for stdout/stderr/stdin..